scrape/scrape.go

```diff
@@ -946,6 +963,10 @@ mainLoop:
 		}
 		last = start
 
+		// Now that we're done scraping. Let's append metadata to remote-write.
+		// Append metadata to remote-write
+		ma.Set(sl.target, sl.cache.ListMetadata())
```

Comment:
Metadata is set incrementally in the cache as it is parsed: first HELP, then Type, etc. Sadly, this makes it impossible to set the metadata as it scraped to remote-write. We need to wait until we're done scraping the target before sending it off the entire metadata cache for this target.